### ISSUE TITLE
ssl configuration problem

### ISSUE THREAD

**Issue by ajoubert** [nginx-proxy/nginx-proxy#288]
Here is what I try to achieve : I want to make sure that both http and https connection work correctly before setting up my other containers.

I ran the image as described : 

```
docker run -d -p 80:80 -p 443:443 --name='proxy' -v /etc/nginx/certs:/etc/nginx/certs -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy
```

I can then access http://localhost and http://myIp on my local network without problem (returning the expected 503 error), but https://localhost returns a "Secure Connection Failed" error, and https://myIp returns a network error.

Turns out my port 443 is only being listened on localhost, not on my local network entrance, and no https connection can initialised.

My problem does not come from my configuration (other containers are able to use port 443 with no problem), but maybe I found where the problem comes from : 

```
docker exec proxy ls /etc/nginx/certs 
```

only return default.cert, and no default.key, which then prevents this part of the code from nginx.tmpl : 

```
{{ if (and (exists "/etc/nginx/certs/default.crt") (exists "/etc/nginx/certs/default.key")) }}
server {
    server_name {{ $host }};
    listen 443 ssl http2 {{ $default_server }};
    access_log /var/log/nginx/access.log vhost;
    return 503;

    ssl_certificate /etc/nginx/certs/default.crt;
    ssl_certificate_key /etc/nginx/certs/default.key;
}
{{ end }}
```

Did anyone else have the same problem ? I'm currently trying to re-generate another .cert and .key to see if it will fix the issue, but being a beginner in docker and ssl, it might take some time.

**Comment by ajoubert** [comment]
I confirm creating a new default.key and default.crt and placing them in /etc/nginx/certs solved my issue, so something went wrong while creating the container ? or is this normal and I missed a step while creating it ?

**Comment by Pimmetje** [comment]
Maybe add this to the documentation more clearly or generate a self signed on first time startup?

**Comment by buchdag** [comment]
This was fixed in #2186
